File: C2/q1-2.cpp
#include<iostream>
#include<string>

using namespace std;

class reverse_string
{
    string str;
    unsigned long length;
public:
    reverse_string(string str) { this->str = str; length = str.length();
        cout << length << endl; 
    };
    string reverse() {
        string str2;
        for (int i = 1; i < length; i++) 
            str2.push_back(str[length-i-1]);
        str2.push_back('\0');
        return str2;
    };
};
            
int main(int argc, char **argv)
{
    string line = argv[1];
    //string line = "Hello Wolrd";
    line.push_back('\0');
    reverse_string rs(line);
    line = rs.reverse();
    cout << line << endl;
    return 0;
} 

File: C2/q1-2.py
import sys

def reverse_string(str):
    """
    this function is gonna reverse string
    """
    reversed_string = ""
    for c in reversed(str):
        if c == '\0':
            None
        else:
            reversed_string += c
    reversed_string += '\0'
    print repr(reversed_string)

def main():
    line = sys.argv[1]
    line += '\0'
    print repr(line)
    reverse_string(line)

if __name__ == "__main__":
    main()

